projects/incubator/src/storefrontlib/cms-components/product/product-images/product-images.component.ts

```diff
@@ -0,0 +1,84 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { ImageGroup, Product } from '@spartacus/core';
+import { CurrentProductService } from '@spartacus/storefront';
+import { BehaviorSubject, combineLatest, Observable, of } from 'rxjs';
+import { distinctUntilChanged, filter, map, tap } from 'rxjs/operators';
+import { ThumbGroup } from '../../../shared/components/image-zoom/product-thumbnails/thumbnails.model';
+
+@Component({
+  selector: 'cx-product-images',
+  templateUrl: './product-images.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ProductImagesComponent {
+  private mainMediaContainer = new BehaviorSubject<ImageGroup>(null);
+
+  protected mainMediaContainer$ = this.mainMediaContainer.asObservable();
+
+  private product$: Observable<
```

Comment:
Can we group these somehow? I believe you grouped them by modifier in `projects/incubator/src/storefrontlib/shared/components/image-zoom/product-view/image-zoom-product-view.component.ts`